src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,44 @@
+package duke;
+
+import duke.command.Command;
+import duke.task.TaskList;
+
+public class Duke {
+    private final Storage storage;
+    private TaskList tasks;
+    private final Ui ui;
```

Comment:
### Should `storage` and `ui` be in uppercase? I noticed this in other areas as well.

Honestly, I wouldn't consider this to be violations, but I'll leave it to you to judge:
- Your variables are final whenever possible, which is good,
- However, that means it probably should be written in uppercase according to the coding convention.
- My personal belief is that only static final variables should be written in uppercase, but the coding convention isn't clear on that.
- However, an example they provided was `final int COLOR_RED = 1;` which leans towards the former,

### I would leave them as non-final to avoid this iffy-ness.

Reply:
Great point. I will probably leave it as non-final as well.